falcon/util/misc.py

```diff
@@ -373,7 +376,13 @@ def secure_filename(filename: str) -> str:
     filename = unicodedata.normalize('NFKD', filename)
     if filename.startswith('.'):
         filename = filename.replace('.', '_', 1)
-    return _UNSAFE_CHARS.sub('_', filename)
+    filename = _UNSAFE_CHARS.sub('_', filename)
+
+    if max_length is not None and len(filename) > max_length:
+        name, ext = os.path.splitext(filename)
+        filename = name[:max_length - len(ext)] + ext
```

Comment:
note that `max_length - len(ext)` may be negative

Reply:
Can you check the logic I have added right now? Do you want to keep ext upto the max_length or remove the concept of max_length entirely incase the ext is longer than max_length?